library/src/scripts/components/Search.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import classNames from "classnames";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { IWithSearchProps, withSearch } from "@library/contexts/SearchContext";
+import { uniqueIDFromPrefix } from "@library/componentIDs";
+import SearchBar from "@library/components/forms/select/SearchBar";
+import { t } from "@library/application";
+import SearchOption from "@library/components/search/SearchOption";
+import { searchClasses } from "@library/styles/searchStyles";
+import { buttonClasses, ButtonTypes } from "@library/styles/buttonStyles";
+
+export interface ICompactSearchProps extends IWithSearchProps, RouteComponentProps<{}> {
+    className?: string;
+    placeholder?: string;
+    buttonClass?: string;
+    showingSuggestions?: boolean;
+    onOpenSuggestions?: () => void;
+    onCloseSuggestions?: () => void;
+    buttonContentClass?: string;
+    cancelContentClassName?: string;
+    theme?: object;
+}
+
+interface IState {
+    query: string;
+}
+
+/**
+ * Implements Compact Search component for header
```

Comment:
Could we update this docblock?

Reply:
I'm also thinking this class name could be more specific. Maybe `IndependentSearch` or `SplashSearch` and moved to the splash directory.